c7n/resources/elasticsearch.py

```diff
@@ -622,3 +622,83 @@ def process(self, resources):
         for r in resources:
             client.update_elasticsearch_domain_config(DomainName=r['DomainName'],
                 DomainEndpointOptions={'EnforceHTTPS': True, 'TLSSecurityPolicy': tls_value})
+
+
+@ElasticSearchDomain.action_registry.register('enable-auditlog')
+class EnableAuditLog(Action):
+
+    """Action to enable audit logs on a domain endpoint
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: enable-auditlog
+                resource: elasticsearch
+                filters:
+                  - type: value
+                    key: 'LogPublishingOptions.AUDIT_LOGS.Enabled'
+                    op: eq
+                    value: false
+                actions:
+                  - type: enable-auditlog
+                    state: True
+    """
+
+    schema = type_schema(
+        'enable-auditlog',
+        state={'type': 'boolean'},
+        required=['state'])
+
+    statement = {
+        "Sid": "OpenSearchCloudWatchLogs",
+        "Effect": "Allow",
+        "Principal": {"Service": ["es.amazonaws.com"]},
+        "Action": ["logs:PutLogEvents", "logs:CreateLogStream"],
+        "Resource": None}
+
+    permissions = (
+        'es:UpdateElasticsearchDomainConfig',
+        'es:ListDomainNames',
+        'logs:DescribeLogGroups',
+        'logs:CreateLogGroup',
+        'logs:PutResourcePolicy')
+
+    def get_loggroup_arn(self, domain):
+        log_group_name = "/aws/OpenSearchService/domains/%s/audit-logs" % (domain)
+        log_group_arn = "arn:aws:logs:{}:{}:log-group:{}:*".format(
+            self.manager.region, self.manager.account_id, log_group_name)
+        return log_group_arn
+
+    def set_permissions(self, log_group_arn):
+        statement = dict(self.statement)
+        statement['Resource'] = log_group_arn
+
+        client = local_session(
+            self.manager.session_factory).client('logs')
+        try:
+            client.create_log_group(logGroupName=log_group_arn.split(":")[-2])
+        except client.exceptions.ResourceAlreadyExistsException:
+            self.log.warning("Log group already exists ...")
+
+        response = client.put_resource_policy(
```

Comment:
if the log group already exists, we would need to add the permission to the extant document to prevent overwrites.

Reply:
@kapilt great catch.. though the policy will only get overwritten if the policy name is same as the existing policy. in your opinion , what would be best way adding another permission block  to the existing policy or add another resource policy with new name ?

Reply:
thanks for catching that .. fixed.